cogs/tmpc.py

```diff
@@ -434,24 +472,80 @@ async def nomod(self, ctx: Context):
         """
         global moderator
 
-        document = await self.check_tmpc_channel(ctx, is_mod=True)
+        document = await self.check_tmpc_channel(ctx.author, ctx.channel.id, is_mod=True)
         await document.voice.set_permissions(moderator.item, overwrite=None)
         await document.chat.set_permissions(moderator.item, overwrite=None)
         embed: Embed = Embed(title="No Mod",
                              description=f"Mods can't see or join this channel anymore")
         await ctx.reply(embed=embed)
 
-    async def check_tmpc_channel(self, ctx: Context, is_mod: bool = False, everyone: bool = False) \
+    @tmpc.command(pass_context=True,
+                  brief="A user to your temp channel",
+                  help="Default add user to the temp channel, for gaming channel the first argument must be 'False'!\n"
+                       "Member must be invited by mentions.\n"
+                       "Limitations:\n"
+                       "- This command can be used 3/hour/user\n"
+                       "- The invite command is limited to 10 user at once")
+    @cooldown(3, 3600, BucketType.user)
+    async def invite(self, ctx: Context, study_channel: bool):
+
+        if study_channel:
+            channel = await self.study_db.find_one({DBKeyWrapperEnum.OWNER.value: ctx.author.id})
+        else:
+            channel = await self.gaming_db.find_one({DBKeyWrapperEnum.OWNER.value: ctx.author.id})
+        channel: Union[GamingChannel, StudyChannel]
+        document = await self.check_tmpc_channel(ctx.author, channel.chat.id)
+
+        overwrites = document.voice.overwrites
+        msg: Message = ctx.message
+        if len(msg.mentions) > 10:
+            await ctx.reply("Warning hit limit")
+
+        changed = False
+        for mention in msg.mentions[:10]:
+            if mention not in overwrites:
+                changed = True
+                overwrites[mention] = PermissionOverwrite(connect=True,
+                                                          view_channel=True)
+        if changed:
+            await document.voice.edit(overwrites=overwrites)
+            await document.chat.edit(overwrites=overwrites)
```

Comment:
Again this will not work since the voice and text channel have different permissions